packages/web/components/Player/Filter/PlayerFilter.tsx

```diff
@@ -122,86 +125,273 @@ export const PlayerFilter: React.FC<Props> = ({
-              borderWidth="2px"
-            />
-            {search.length > 0 && (
-              <InputRightElement>
-                <IconButton
-                  p="2"
-                  variant="link"
-                  colorScheme="white"
-                  icon={<CloseIcon />}
-                  onClick={() => {
-                    setSearch('');
-                    setQueryVariable('search', '%%');
+          <Box
+            flexGrow={{
+              base: '2', // expand to take up more space when side-by-side with filters button
+              md: '0',
+            }}
+            marginRight={{
+              base: '4',
+              md: '0',
+              '2xl': '6',
+            }}
+            marginBottom={{
+              base: '0',
+              md: '4',
+              '2xl': '0',
+            }}
+            maxW={{
+              base: 'sm',
+              md: '100%',
+              '2xl': 'sm',
+            }}
+          >
+            {/**
+             * SEARCH BOX
+             * Initial minWidth of the search input is 14em, so it will fit on a tiny screen.
+             * After md breakpoint, the minwidth is 30em. (Closer match to the dropdowns while in a stacked layout)
+             * After 2xl breakpoint, the minWidth is 20em so it will fit in a single row with the filter dropdowns.
+             * There's no search button any more; it's set up to search onChange once the search string is longer than 2 chars.
+             */}
+            <Form width="fill" onSubmit={onSearch} display="flex">
+              <InputGroup>
+                <InputLeftElement
+                  pointerEvents="none"
+                  children={<SearchIcon />}
+                />
+                <Input
+                  background="dark"
+                  borderColor="borderPurple"
+                  borderLeftRadius={4}
+                  borderRightRadius={{
+                    base: '4',
+                    md: '0',
                   }}
-                  aria-label="Clear Search"
+                  borderWidth={{
+                    base: '1px',
+                    md: '2px',
+                  }}
+                  fontSize={{
+                    base: 'sm',
+                    md: '1em',
+                  }}
+                  height="40px"
+                  minW={{
+                    base: '14em',
+                    md: '30em',
+                    '2xl': '20em',
+                  }}
+                  onChange={(e) => {
+                    setSearch(e.target.value);
+                    if (e.target.value.length >= SEARCH_STR_MIN_LENGTH) {
+                      setQueryVariable('search', `%${e.target.value}%`); // If using setQueryVariable('search', `%${search}%`) and/or search.length, it misses the last character entered i.e. `vid` comes out as `vi`
+                    } else {
+                      setQueryVariable('search', `%%`);
+                    }
+                  }}
```

Comment:
```suggestion
                  onChange={({ target: { value } }) => {
                    setSearch(value);
                    if (value.length >= MIN_SEARCH_LENGTH) {
                      setQueryVariable('search', `%${value}%`); // If using setQueryVariable('search', `%${search}%`) and/or search.length, it misses the last character entered i.e. `vid` comes out as `vi`
                    } else {
                      setQueryVariable('search', `%%`);
                    }
                  }}
```

Reply:
`{ target: { value } }` was a good one, do you have a link or search query for me to read more on that?